test/recipes/30-test_ech.t

```diff
@@ -0,0 +1,21 @@
+#! /usr/bin/env perl
+# Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+# Copyright (c) 2022, Oracle and/or its affiliates.  All rights reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use OpenSSL::Test::Utils;
+use OpenSSL::Test qw/:DEFAULT srctop_file srctop_dir bldtop_dir bldtop_file/;
+
+setup("test_ech");
+
+plan skip_all => "ECH tests not supported in this build"
+    if !disabled("fips") || disabled("ech") || disabled("tls1_3") || disabled("ec") || disabled("tls1_2") || disabled("ecx");
```

Comment:
Why this long list of things that cannot be disabled for us to run any tests?

Reply:
it's built up over time (with the not-yet-in-a-PR code) - I've not tried a FIPS only ECH build (and not sure that's realistic today), we shouldn't try run these tests in a no-ech build, we need tls1.3 and ec/ecx (for HPKE) and we need tls1.2 but only for the dodgy test mentioned above related to `OPENSSL_ECH_ALLOW_CUST_INJECT`

if there's a more terse way to do that, that's fine by me

Reply:
"enable-fips" doesn't give you a "FIPS only build". It just means the FIPS provider is built. Unless you explicitly configure the fips provider for a test it won't be used so it shouldn't impact the test. (Although it would be kind of interesting to know whether ECH is compatible with the FIPS provider - but that can be a question for later)

Reply:
I've reduced as below for now. Guess we'll see what the CI makes of that:
```
if disabled("ech") || disabled("tls1_3") || disabled("ec") || disabled("ecx");
```